src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -238,15 +242,72 @@ def validate(s: str) -> str:
             column_value = self.column_joiner.join(sorted(set(column_value)))
             row[column_name] = column_value
 
+    def _get_header(self, object_key):
+        try:
+            header = self.storage_service.head(object_key=object_key)
+        except self.storage_service.client.exceptions.ClientError as e:
+            if int(e.response['Error']['Code']) == 404:
+                return None
+            else:
+                raise e
+        else:
+            return header
+
+    @classmethod
+    def _get_seconds_until_expire(cls, header: dict) -> float:
+        """
+        Get the number of seconds before a cached manifest is past its expiration.
+        :param header: A storage service object header dict
+        :return: time to expiration in seconds
+        """
+        # example x-amz-expiration = 'expiry-date="Fri, 21 Dec 2012 00:00:00 GMT", rule-id="Rule for testfile.txt"'
+        date_time_now = datetime.now(timezone.utc)
+        x_amz_expiration = parse_dict_header(header['x-amz-expiration'])
+        expiry_date = parse_date(x_amz_expiration['expiry-date']).replace(tzinfo=date_time_now.tzinfo)
+        seconds_until_expire = (expiry_date - date_time_now).total_seconds()
+        # Verify value calculated from 'x-amz-expiration' matches value calculated from 'LastModified'
+        last_modified = header['LastModified']
+        assert isinstance(last_modified, datetime)
+        expected_expiry_date = last_modified + timedelta(days=config.manifest_expiration)
+        expected_value = (expected_expiry_date - date_time_now).total_seconds()
+        date_diff_margin = 10  # seconds
+        if abs(seconds_until_expire - expected_value) > date_diff_margin:
+            expected_expiry_date = date_time_now + timedelta(seconds=seconds_until_expire)
+            logger.error('x-amz-expiration %s != calculated %s', expiry_date, expected_expiry_date)
+        else:
+            pass
```

Comment:
Hmm, I think this is taking it too far. Unless this is a oversight that you intended to remove, I will probably have to specify more clearly where I think redundant `else` branches should be added.